diff --git a/node_modules/@fastify/reply-from/lib/utils.js b/node_modules/@fastify/reply-from/lib/utils.js
index e98a6b0..462f5f7 100644
--- a/node_modules/@fastify/reply-from/lib/utils.js
+++ b/node_modules/@fastify/reply-from/lib/utils.js
@@ -57,6 +57,12 @@ function stripHttp1ConnectionHeaders (headers) {
 // issue ref: https://github.com/fastify/fast-proxy/issues/42
 function buildURL (source, reqBase) {
   let baseOrigin = reqBase ? new URL(reqBase).href : undefined
+
+  // A fix until https://github.com/fastify/fastify-reply-from/pull/319 is merged
+  if (baseOrigin && source.startsWith('//')) {
+    source = '.' + source;
+  }
+
   const dest = new URL(source, reqBase)
 
   // if base is specified, source url should not override it
